packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -267,19 +268,20 @@ class _DevFSHttpWriter {
       HttpClientRequest request = await _client.putUrl(httpAddress);
       request.headers.removeAll(HttpHeaders.ACCEPT_ENCODING);
       request.headers.add('dev_fs_name', fsName);
+      // TODO(goderbauer): transfer real Uri (instead of file path) when remote end supports it
```

Comment:
I currently don't have the intention to jump in the Dart VM as there are other things I need to fix to make Flutter work for Windows. However, I am going to file an issue linking to the code bits here to make sure this doesn't get forgotten.

Do you want to do the changes on the VM side? :)